src/main/java/Duke/Interpreter.java

```diff
@@ -0,0 +1,247 @@
+package Duke;
+
+import Duke.Exceptions.*;
+import Duke.Tasks.*;
+import java.lang.String;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * The Duke.Interpreter interpreters the users' input
+ */
+public class Interpreter {
+    private final static DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private static enum Operation {
+        // all the operations
+        mark, list, delete, deadline, event, bye, todo, unmark, find
+    }
+
+    // format the time in the form of 11/10/2019 5pm
+
+    /**
+     * The method mark marks the Task as done
+     * @param command the command by the user
+     * @param table the task table
+     * @return Task
+     * @throws InvalidCommandException
+     * @throws OutRangeException
+     */
+    public static Task mark(String command, TaskTable table) throws InvalidCommandException, OutRangeException {
+        int len = table.size();
+        int index;
```

Comment:
The Java coding standard says to initialize variables where they are declared. Would it be better to declare index at line 39 instead?